### DISCUSSION TITLE
What do you think about linking common error messages to the relevant section in the project?

### DISCUSSION THREAD

**Discussion by levibostian** [dimasusername/ConcurrencyRecipes#9] [Ideas]
Thanks a lot, Matt, for starting this project! 

I had an idea to share to see if it could help others who are trying to fix all of their compiler warnings. 

### My expected behavior
* Enable compiler concurrency checking. Compile my project in Xcode. 
* Xcode gives me a warning. Example: "passing argument of non-sendable type '___' outside of main actor-isolated context may introduce data races". 
* I go to this project, quickly find recipes that directly help resolve this particular warning. 

### My actual behavior 
* ...
* I go to this project, open each document one-by-one, scroll through each doc trying to find the section relevant to this warning. It's a manual process to guess what recipe may help with this warning. 

If the README had a section in it that had a list of common Xcode errors and hyperlinks to relevant sections in the docs, perhaps that could improve navigation for people like me?

**Comment by mattmassicotte** [comment]
I really love this idea! I know that the warnings/errors are going to evolve, so it may be tricky to get this stable. But I still think its a worth-while thing to try to do.

**Comment by disbelief** [comment]
I was scouring the internet for just such a resource. Copy-pasting the concurrency warnings into Google and StackOverflow has yielded poor results.

With the recent launch of Xcode 16, I expect many devs are experiencing the same frustration. Would this repo be the right place for such an index? Perhaps a wiki page(s) that link to the specific recipe?

**Comment by mattmassicotte** [comment]
I forgot about this! But, it's a really good idea.

I think the intention of the migration guide on Swift.org was to cover errors more directly. It's nice, because it includes the error messages, the code that caused them, and some explanation on what is actually going on. It's definitely more work to put stuff there, but I think it will have way more visibility. And, I also think explaination are critical. There isn't always a simple fix, and developing a deeper understanding of why the error is happening can be really important.
